### PR TITLE
[Ready for Code Review] Q&A 3.0.0

### PR DESCRIPTION
**Pull request by bricesanchez**
- [x] Clean useless files
- [x] Update locale picker in forms
- [x] Fix tree reordering
- [x] Use config for backend route
- [x] Improve engine inclusion
- [x] Whitelist assets
- [x] Fix gallery selected option
- [x] Use new icons set
- [x] Fix gallery cancel link
- [x] Check tests
- [x] Add ability to update translated slug

Could you do a code review and merge it if it's good?

### CODE REVIEW

File: app/views/refinery/portfolio/admin/galleries/_gallery.html.erb [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+  image_add_url = refinery.new_portfolio_admin_gallery_item_path(gallery)
+  new_child_add_url = refinery.new_portfolio_admin_gallery_path(:parent_id => gallery.id)
+  preview_url = refinery.portfolio_gallery_path(gallery)
+  edit_url = refinery.edit_portfolio_admin_gallery_path(gallery)
+  delete_url = refinery.portfolio_admin_gallery_path(gallery)
+  delete_options = {
+     class: "cancel confirm-delete",
+            data: {confirm: t('message',  scope: 'refinery.admin.delete', title: gallery.title)}
+          }
+%>
```
Comment (parndt) [inline app/views/refinery/portfolio/admin/galleries/_gallery.html.erb]: the indentation here seems dubious

File: app/views/refinery/portfolio/admin/items/_item.html.erb [hunk 2]
```diff
@@ -1,27 +1,39 @@
+<% # setup params for various action links
+  preview_url = gallery_path_for(item)
+  edit_url = refinery.edit_portfolio_admin_item_path(item)
+  delete_url = refinery.portfolio_admin_item_path(item)
+  delete_options = {
+     class: "cancel confirm-delete",
+            data: {confirm: t('message',  scope: 'refinery.admin.delete', title: item.title)}
```
Comment (parndt) [inline app/views/refinery/portfolio/admin/items/_item.html.erb]: strange indentation

File: app/views/refinery/portfolio/admin/items/_item.html.erb [hunk 3]
```diff
@@ -1,27 +1,39 @@
...
+              <% if translation.title.present? %>
+                <%= link_to refinery.edit_portfolio_admin_item_path(item, :switch_locale => translation.locale), :class => 'locale', title: translation.locale.upcase do %>
+
+                  <div class="<%=translation.locale %> locale_marker">
+                    <%= locale_text_icon(translation.locale.upcase) %>
+                  </div>
+                <% end %>
+              <% end %>
+            <% end %>
+          </span>
```
Comment (parndt) [inline app/views/refinery/portfolio/admin/items/_item.html.erb]: very similar code exists already here https://github.com/refinery/refinerycms-portfolio/pull/103/files#diff-59c4eb32a2e806eecaf667bf94b56e04R20

Comment (bricesanchez) [inline app/views/refinery/portfolio/admin/items/_item.html.erb]: Yes we should refactor this portion of code to use a partial available in refinerycms-i18n.

It's the same portion of code in all of our admin index...

Comment (parndt) [inline app/views/refinery/portfolio/admin/items/_item.html.erb]: fine :)

**Comment by bricesanchez** [comment]
Is someone could review and merge this PR ?

**Comment by parndt** [comment]
@bricesanchez could you revert the indentation back to what it was?

i.e.:

```ruby
protected

def some_method
end

private

def another_method
end
```

Instead of:

```ruby
protected

  def some_method
  end

private

  def another_method
  end
```

**Comment by bricesanchez** [comment]
It's ready @parndt :)

**Comment by parndt** [comment]
Thanks!

**Pull request merged**